### PR TITLE
[#IR-211] Show disabled SMT for members who participated in the task in other sprints

### PR DESCRIPTION
**Pull request by himanshu-garg-165**

### CODE REVIEW

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts [hunk 1]
```diff
@@ -255,7 +282,9 @@ export class RetrospectTaskModalComponent implements OnDestroy {
                 valueFormatter: (cellParams) => {
                     return MEMBER_TASK_ROLES_LABEL[cellParams.value];
                 },
-                editable: isSprintEditable,
+                editable: (params: IsColumnFuncParams) => {
```
Comment (chetanyakan) [inline src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts]: Either specify the type of params for all columns or don't specify the type at all to keep things consistent.

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts [hunk 2]
```diff
@@ -169,8 +180,25 @@ export class RetrospectTaskModalComponent implements OnDestroy {
...
+        this.gridApi.showLoadingOverlay();
+        // To clear the current content of the grid
+        this.gridApi.setRowData(null);
+        const getTaskMemberSummary$ = this.getSprintTaskMemberSummary(true);
+        getTaskMemberSummary$.subscribe(
+            () => {
+                this.gridApi.hideOverlay();
+            },
+            () => {
+                this.gridApi.hideOverlay();
```
Comment (chetanyakan) [inline src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts]: Use complete callback if we have to perform an action in both next and error callbacks.

Comment (anmolghosh) [reply src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts]: `() => this.gridApi.hideOverlay()`

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts (hunk 1 above)
Comment (anmolghosh) [reply src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts]: we should specify the type as long as there are no other concerns

Comment (chetanyakan) [reply src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts]: Added elsewhere, except for GridReadyEvent for which seems to be a type mismatch.

File: src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts [hunk 3]
```diff
@@ -322,14 +322,14 @@ export class RetrospectTaskModalComponent implements OnDestroy, AfterViewChecked
             {
                 headerName: 'Sprint Points',
                 field: 'SprintPoints',
-                editable: (params) => {
+                editable: (params: IsColumnFuncParams) => {
                     return isSprintEditable && params.data.Editable;
```
Comment (himanshu-garg-165) [inline src/app/retrospective/retrospect-task-modal/retrospect-task-modal.component.ts]: @chetanyakan Please make sure you properly test the changes you make. Changing the flag name from `Editable` to `Current` is not done in all the places.

**Pull request merged**